plugins/juju/juju.plugin.zsh

```diff
@@ -0,0 +1,134 @@
+#   - `ds` suffix means --destroy-storage                    #
+# ---------------------------------------------------------- #
+alias jam='juju add-model --config logging-config="<root>=WARNING; unit=DEBUG"\
+  --config update-status-hook-interval="60m"'
+alias jb='juju bootstrap'
+alias jbm='juju bootstrap microk8s'
+alias jc='juju config'
+alias jdc='juju destroy-controller --destroy-all-models'
+alias jdc!='juju destroy-controller --destroy-all-models --force --no-wait -y'
+alias jdcds='juju destroy-controller --destroy-all-models --destroy-storage'
+alias jdcds!='juju destroy-controller --destroy-all-models --destroy-storage --force --no-wait -y'
+alias jdm='juju destroy-model'
+alias jdm!='juju destroy-model --force --no-wait -y'
+alias jdmds='juju destroy-model --destroy-storage'
+alias jdmds!='juju destroy-model --destroy-storage --force --no-wait -y'
+alias jde='juju deploy --channel=edge'
+alias jd='juju deploy'
+alias jdl='juju debug-log --ms'
+alias jdlr='juju debug-log --ms --replay'
+alias jcon='juju consume'
+alias jeb='juju export-bundle'
+alias jex='juju expose'
+alias jh='juju help'
+alias jkc='juju kill-controller -y -t 0'
+alias jm='juju models'
+alias jmc='juju model-config'
+alias jof='juju offer'
+alias jra='juju run-action'
+alias jraw='juju run-action --wait'
+alias jrel='juju relate'
+alias jrm='juju remove-application'
+alias jrm!='juju remove-application --force --no-wait'
+alias jrmds='juju remove-application --destroy-storage'
+alias jrmds!='juju remove-application --destroy-storage --force --no-wait'
+alias jrmrel='juju remove-relation'
+alias jrmsas='juju remove-saas'
+alias jrp='juju refresh --path'
+alias jrs='juju remove-storage'
+alias jrs!='juju remove-storage --force'
+alias jsa='juju scale-application'
+alias jsh='juju ssh'
+alias jshc='juju ssh --container'
+alias jshm='juju show-model'
+alias jstj='juju status --format=json'
+alias jst='juju status --relations --storage --color'
+alias jsw='juju switch'
+
+# ---------------------------------------------------------- #
+# Functions (in alphabetic order)                            #
+# ---------------------------------------------------------- #
+
+# Get app or unit address
+jaddr() {
+    # $1 = app name
+    # $2 = unit number (optional)
+    
+    if ! command -v jq &> /dev/null
+    then
+        echo "jq is required but could not be found."
+        return 1
+    fi
+
+    
+    if [ "$#" -eq 1 ]; then
+        # Get app address
+        juju status "$1" --format=json | \
+          jq -r ".applications.\"$1\".address"
+        
+    elif [ "$#" -eq 2 ]; then
+        # Get unit address
+        juju status "$1/$2" --format=json | \
+          jq -r ".applications.\"$1\".units.\"$1/$2\".address"
+        
+    else
+        echo "Invalid number of arguments."
+        echo "Usage:   jaddr <app-name> [<unit-number>]"
+        echo "Example: jaddr karma"
+        echo "Example: jaddr karma 0"
+        return 1
+    fi
+}
```

Comment:
Look up the code style guide in the wiki: 2-space indentation, `[[` instead of `[`, `then` always in the same line as the `if`. Also it's cleaner to add the pipe character on the next line:
```zsh
juju status "$1" --format=json \
  | jq -r ".applications.\"$1\".address"
```

Also, are those `\"` supposed to go there? I think it doesn't make a valid jq selector, but I haven't tested it.

Reply:
without the `\"`, dashes in application name are not ingested correctly